service/crt/service.go

```diff
@@ -193,8 +192,9 @@ func (s *Service) newCertificateListWatch() *cache.ListWatch {
 				return nil, err
 			}
 
-			watcher := watch.NewStreamWatcher(&k8sutil.CertificateDecoder{
-				Stream: stream,
+			watcher := watch.NewStreamWatcher(&certificateDecoder{
+				decoder: json.NewDecoder(stream),
+				close:   stream.Close,
```

Comment:
That is maybe a bit cleaner but I don't have a strong opinion on it. We should move to the operatorkit support once its ready.